src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -17,108 +19,760 @@
 
 namespace Microsoft.AspNetCore.Mvc.Internal
 {
-    public class ControllerActionInvoker : FilterActionInvoker
+    public class ControllerActionInvoker : IActionInvoker
```

Comment:
Virtually all of this is copy-paste. The other thing I really changed here was the number and ordering of the constructor arguments.